rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -86,9 +86,50 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
             replicate_last_batch_to_pad_partial_shard();
         }
     }
-    //shuffle dataset if set
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
+    if (_meta_data_reader)
```

Comment:
I'm still thinking whether we have to add it as a reader argument or whether is there a way to fetch the meta_data type from meta reader to only do it in case of Polygon meta type. Will make the change after some discussion with Fiona